services/net/transcription/TranscriptionManager.cs

```diff
@@ -76,12 +91,24 @@ public override async Task RunAsync()
                         var tps = String.Join(',', topics);
                         this.Logger.LogInformation("Consuming topics: {tps}", tps);
 
-                        // TODO: Need to learn how to safely stop listening without losing content.  Every time a service goes down it will most likely lose content.
-                        await this.Consumer.ListenAsync<string, SourceContent>(HandleMessageAsync, topics);
+                        if (!this.Consumer.IsReady) this.Consumer.Open();
+
+                        // TODO: Not sure if the consumer should stop before changing its subscription.
+                        this.Consumer.Subscribe(topics);
+
+                        // Create a new thread if the prior one isn't running anymore.
+                        if (_consumer == null || _notRunning.Contains(_consumer.Status))
+                        {
+                            _consumer = Task.Factory.StartNew(() => ConsumerHandler());
+                        }
 
                         // Successful run clears any errors.
                         this.State.ResetFailures();
                     }
+                    else if (topics.Length == 0)
+                    {
+                        this.Consumer.Stop();
```

Comment:
Make sure this doesn't cause issues if there are no topics configured.

Reply:
Not sure how I would remedy this.

Reply:
If there are no topics, then it'll attempt to stop the consumer.  You can test this and see what happens.  If no error is thrown, then you don't have to do anything.  If an error is thrown then we'll need to figure out a way to handle this.